iocage_lib/ioc_list.py

```diff
@@ -364,12 +396,36 @@ def list_all(self, jails):
 
                         ui_data = json.load(u)
                         admin_portal = ui_data["adminportal"]
-                        admin_portal = ','.join(
-                            map(
-                                lambda v: admin_portal.replace('%%IP%%', v),
-                                all_ips
+                        admin_portals = []
+                        for portal in admin_portal.split(','):
+                            if conf.get('nat'):
+                                portal_uri = urllib.parse.urlparse(portal)
+                                portal_port = portal_uri.port or 80
+                                # We do this safely as it's possible
+                                # dev hasn't added it to plugin's json yet
+                                nat_port = nat_forwards_dict.get(portal_port)
+                                uri = ''
+                                if portal_uri.scheme:
+                                    uri = f'{portal_uri.scheme}://'
+                                uri += portal_uri.hostname
+                                if nat_port:
+                                    uri += f':{nat_port}'
+                                if portal_uri.path:
+                                    uri += portal_uri.path
```

Comment:
You're missing the other parts, like query.

Why not replace portal_uri.port alone and rebuild the uri with urlunparse?